lib/src/config_writer/vscode_config_writer.dart

```diff
@@ -33,34 +33,49 @@ class VscodeConfigWriter extends ConfigWriter {
   String writeConfig(String fileContent) {
     /// launch.json usually contains comments, which is valid only in JSON5.
     /// At this point however we cannot preserve these comments.
-    fileContent = fileContent.replaceAll(RegExp('.+//.+\n'), "");
-
-    var configJson = jsonDecode(fileContent);
-
-    var configList = (configJson['configurations'] as Iterable);
-
-    if (configName != null) {
-      configList = configList.where((config) => config['name'] == configName);
+    try {
+      Map<String, dynamic> configJson = jsonDecode(fileContent);
+      var configs = (configJson['configurations'] as List);
+
+      List<String> dartDefineList = getDartDefineList();
+
+      if (configName != null) {
+        var namedConfig = configs
+            .where((config) => config['name'] == configName)
+            .toList()
+            .first;
+        var updatedNamedConfig = updateConfig(namedConfig, dartDefineList);
+        configs.removeWhere((element) => element['name'] == configName);
+        configs.add(updatedNamedConfig);
+        configJson['configurations'] = configs;
+      } else {
+        configJson['configurations'] = configs
+            .map((configMap) => updateConfig(configMap, dartDefineList))
+            .toList();
+      }
+      return prettifyJson(configJson);
+    } catch (err) {
+      print(
+          'Exception occurred: Possible error is comments in JSON. Additional information: $err');
+      return fileContent;
     }
-
-    var dartDefineList = getDartDefineList();
-
-    configJson['configurations'] =
-        configList.map((configMap) => updateConfig(configMap, dartDefineList));
-
-    return prettifyJson(configJson);
   }
 
   /// Update a single VS Code [config] with [dartDefineList].
   Map<String, dynamic> updateConfig(
     Map<String, dynamic> config,
-    Iterable<String> dartDefineList,
+    List<String> dartDefineList,
   ) {
-    return config.update(
-      'args',
-      (value) => getNonDartDefineArguments(value).followedBy(dartDefineList),
-      ifAbsent: () => dartDefineList,
-    );
+    Map<String, dynamic> configUpdate = Map<String, dynamic>.from(config);
+
+    print('ConfigUpdate: $configUpdate');
```

Comment:
Can you remove all the print statements?

Reply:
Sure thing, Thought I caught them all. My bad